src/main.ts

```diff
@@ -0,0 +1,89 @@
+import * as core from '@actions/core'
+import * as logger from './actions/logger'
+import * as runCli from './actions/run'
+import * as utils from './actions/utils'
+import {getJobInfo, setJobInfoEnvVar} from './actions/job'
+import {FRAMEWORK_TYPES} from './constants'
+import {Octokit} from '@octokit/action'
+import {validateInputs} from './actions/inputs'
+
+const apiKey: string = core.getInput('api_key', {required: true})
+const testFramework: string = core.getInput('test_framework', {
+    required: false
+})
+const testPath: string[] = core.getMultilineInput('test_path', {
+    required: false
+})
+const coverageFramework: string = core.getInput('coverage_framework', {
+    required: false
+})
+const coveragePath: string[] = core.getMultilineInput('coverage_path', {
+    required: false
+})
+const actionDisabled: boolean = core.getBooleanInput('disable_action', {
+    required: false
+})
+const cliVersion: string = core.getInput('cli_version', {required: false})
+
+validateInputs(
+    testFramework,
+    testPath,
+    coverageFramework,
+    coveragePath,
+    actionDisabled
+)
+const octokit: Octokit = new Octokit()
+
+async function run(): Promise<void> {
+    try {
+        logger.info('Start gettin job info...')
+        const jobInfo = await getJobInfo(octokit)
+        logger.info(`jobInfo: ${jobInfo.id}, ${jobInfo.name}`)
+        if (!jobInfo.id || !jobInfo.name) {
+            logger.notice("Workflow job information couldn't retrieved!")
+        }
+        await setJobInfoEnvVar(jobInfo)
+        logger.info(`Env vars set!`)
```

Comment:
For the cases which we couldn't get job info because of insufficient permissions, who get `RUNNER_NAME` and current time to be sent to backend so we can try to get job info in our backend later over our Github app installation? CLI itself?

Reply:
They are sent by CLI so we do not need to do anything here.